atomic_reactor/plugins/exit_delete_from_registry.py

```diff
@@ -42,64 +42,163 @@ def __init__(self, tasker, workflow, registries):
 
         self.registries = deepcopy(registries)
 
+    def setup_secret(self, registry, secret_path):
+        auth = None
+
+        if secret_path:
+            self.log.debug("registry %s secret %s", registry, secret_path)
+            dockercfg = Dockercfg(secret_path).get_credentials(registry)
+            try:
+                username = dockercfg['username']
+                password = dockercfg['password']
+            except KeyError:
+                self.log.error("credentials for registry %s not found in %s",
+                               registry, secret_path)
+            else:
+                self.log.debug("found user %s for registry %s", username, registry)
+                auth = requests.auth.HTTPBasicAuth(username, password)
+
+        return auth
+
+    def request_delete(self, url, manifest, insecure, auth):
+        response = requests.delete(url, verify=not insecure, auth=auth)
+
+        if response.status_code == requests.codes.ACCEPTED:
+            self.log.info("deleted manifest %s", manifest)
+            return True
+        elif response.status_code == requests.codes.NOT_FOUND:
+            self.log.warning("cannot delete %s: not found", manifest)
+        elif response.status_code == requests.codes.METHOD_NOT_ALLOWED:
+            self.log.warning("cannot delete %s: image deletion disabled on registry",
+                             manifest)
+        else:
+            msg = "failed to delete %s: %s" % (manifest, response.reason)
+            self.log.error("%s\n%s", msg, response.text)
+            raise PluginFailedException(msg)
```

Comment:
this is also copied-n-pasted from the original code (as part of the code movement).  I can look to add coverage here.